CHANGES.md

```diff
@@ -1,6 +1,430 @@
+  - added support for an experimental 'TOML' module
+  - added support for a distributed guided and dynamic iterators
+  - improved support for mixing MPI with ugni, gasnet/aries, and/or qthreads
+  - improved Chapel-Python interoperability for the 'ZMQ' (ZeroMQ) module
+  - continued to improve the 'LinearAlgebra' module
+  - converted many library routines to 'throw' in the event of errors
+* performance improvements:
+  - added support for dynamic array registration with the network stack
+  - significantly improved the ability of the LLVM back-end to optimize Chapel
+  - reduced the amount of locking used for associative array accesses
+  - added the ability to serialize records, permitting local copy optimizations
+  - closed a few library-based memory leaks
+* array / domain / domain map improvements:
+  - generalized LayoutCSR to LayoutCS to support CSR and CSC sparse layouts
+  - improved the locality properties of sparse Block-distributed domains/arrays
+  - improved the locality of distributed rank-change and reindex expressions
+  - improved the behavior and power of the 'Replicated' distribution
+* other improvements:
+  - various improvements for working with C pointers and memory
+  - various improvements for the 'ugni' communication layer on Cray systems
+  - numerous bug fixes, error message improvements, and third-party updates
+
+New Dependences / Configuration Changes
+---------------------------------------
+* added a new 'configure'/'make install' option for building+installing Chapel
+  (see http://chapel.cray.com/docs/latest/usingchapel/building.html#installing-chapel)
+* users of the LLVM-based front- or back-ends must now have CMake to build
+* users of the `Regexp` module / RE2 must now have a C++11 compiler to build
+* made 'printchplenv' distinguish between config-file- vs. env-set variables
+* made 'printchplenv' infer the location of CHPL_HOME
+
+New Tools / Tool Changes
+------------------------
+* added an initial version of 'Mason' v0.1.0, Chapel's package manager
+  (see http://chapel.cray.com/docs/master/tools/mason/mason.html)
+* added a new 'c2chapel' tool that converts C headers to Chapel extern decls
+  (see http://chapel.cray.com/docs/master/tools/c2chapel/c2chapel.html)
+* 'chpl'-generated executables now take the main module name rather than 'a.out'
+* added support for LaTeX in chpldoc comments via mathjax
+* made chpldoc issue a warning if it detects open/close comment mismatches
+
+Semantic Changes / Changes to Chapel Language
+---------------------------------------------
+* made explicit 'module' declarations use strict error handling by default
+  (see TODO)
+* made reduce intents preserve the initial value of the reduction variable
+  (see TODO)
+* added support for iterators that can yield void values
+  (see http://chapel.cray.com/docs/latest/technotes/voidVariables.html)
+* relaxed the requirement that iterators must contain `yield` statements
+  (see TODO??)
```

Comment:
http://chapel.cray.com/docs/latest/technotes/voidVariables.html is updated to include non-yielding iterators now.  Section 21.2 of the spec (as far back as 1.13.0) also says, "An iterator need not contain any yield statements."

Reply:
Thanks for following up David, I saw that PR go in and added the link to my working copy.  Did I get it wrong that the compiler only just started permitting yield-less iterators in this release?  Or have we just not conformed to the spec in this regard?

Reply:
You're right that it has only started working this release.  The spec just said that it should work as far back as 1.13.

Reply:
OK great, I'm moving this out of semantic changes then and into error messages / semantic checks.  Thanks for pointing that out.